src/script/lua_api/l_env.cpp

```diff
@@ -465,14 +482,29 @@ int ModApiEnv::l_dig_node(lua_State *L)
 		lua_pushboolean(L, false);
 		return 1;
 	}
+
+	if (lua_gettop(L) >= 2 && (! lua_isnoneornil(L, 2))) {
```

Comment:
BTW this line of code needs investigation; theoretically `lua_isnoneornil` should be enough, but somehow I have to check for the number of args here